zkevm-circuits/src/evm_circuit/witness.rs

```diff
@@ -559,9 +560,41 @@ impl Rw {
 
 impl From<&bus_mapping::circuit_input_builder::ExecStep> for ExecutionState {
     fn from(step: &bus_mapping::circuit_input_builder::ExecStep) -> Self {
-        // TODO: error reporting. (errors are defined in
-        // circuit_input_builder.rs)
-        debug_assert!(step.error.is_none());
+        if let Some(error) = step.error.as_ref() {
+            let state = match error {
+                ExecError::ExecutionReverted => ExecutionState::REVERT,
+                ExecError::InvalidOpcode => ExecutionState::ErrorInvalidOpcode,
+                ExecError::StackOverflow => ExecutionState::ErrorStackOverflow,
+                ExecError::StackUnderflow => ExecutionState::ErrorStackUnderflow,
+                ExecError::WriteProtection => ExecutionState::ErrorWriteProtection,
+                ExecError::Depth => ExecutionState::ErrorDepth,
+                ExecError::InsufficientBalance => ExecutionState::ErrorInsufficientBalance,
+                ExecError::ContractAddressCollision => {
+                    ExecutionState::ErrorContractAddressCollision
+                }
+                ExecError::InvalidCode => ExecutionState::ErrorInvalidCreationCode,
+                ExecError::InvalidJump => ExecutionState::ErrorInvalidJump,
+                ExecError::ReturnDataOutOfBounds => ExecutionState::ErrorReturnDataOutOfBound,
+                ExecError::CodeStoreOutOfGas => ExecutionState::ErrorOutOfGasCodeStore,
+                ExecError::MaxCodeSizeExceeded => ExecutionState::ErrorMaxCodeSizeExceeded,
+                ExecError::OutOfGas(oog_error) => match oog_error {
+                    OogError::Constant => ExecutionState::ErrorOutOfGasConstant,
+                    OogError::PureMemory => ExecutionState::ErrorOutOfGasPureMemory,
+                    OogError::Sha3 => ExecutionState::ErrorOutOfGasSHA3,
+                    OogError::CallDataCopy => ExecutionState::ErrorOutOfGasCALLDATACOPY,
+                    OogError::CodeCopy => ExecutionState::ErrorOutOfGasCODECOPY,
+                    OogError::ExtCodeCopy => ExecutionState::ErrorOutOfGasEXTCODECOPY,
+                    OogError::ReturnDataCopy => ExecutionState::ErrorOutOfGasRETURNDATACOPY,
+                    OogError::Log => ExecutionState::ErrorOutOfGasLOG,
+                    OogError::Call => ExecutionState::ErrorOutOfGasCALL,
+                    OogError::CallCode => ExecutionState::ErrorOutOfGasCALLCODE,
+                    OogError::DelegateCall => ExecutionState::ErrorOutOfGasDELEGATECALL,
+                    OogError::Create2 => ExecutionState::ErrorOutOfGasCREATE2,
+                    OogError::StaticCall => ExecutionState::ErrorOutOfGasSTATICCALL,
+                },
+            };
```

Comment:
I re-think about it and feel that we'd better have 
```
impl From<ExecError> for ExecutionState
```

to make the codes cleaner